dali/kernels/reduce/reduce.h

```diff
@@ -0,0 +1,453 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_REDUCE_REDUCE_H_
+#define DALI_KERNELS_REDUCE_REDUCE_H_
+
+#include <cassert>
+#include <utility>
+#include <vector>
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/core/format.h"
+#include "dali/core/small_vector.h"
+#include "dali/core/span.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/core/tensor_view.h"
+#include "dali/core/util.h"
+
+namespace dali {
+namespace kernels {
+namespace reductions {
+
+struct identity {
```

Comment:
These shall be upper-case https://google.github.io/styleguide/cppguide.html#Type_Names